### ISSUE TITLE
How do you use this with state management libraries such as bloc

### ISSUE THREAD

**Issue by becjit** [esentis/infinite_grouped_list#3]
Hi,  first of all thanks for the package. I am a fan of your multiple search selection package. I am not quite able to wrap my head around how to use this with statemanagement solutions such as bloc. It seems to me call to fetch the data and loading the data both are tightly coupled in onLoadmore. 
Ideally onloadmore could have had a option of fireing an event and the data could have loaded through controller.
Let me know if I am missing something obvious

**Comment by esentis** [comment]
Hello there @becjit thanks for your kind words ! 
So the bottom line, you would like a way of triggering the `onLoadMore` through the controller ? If I'm missing something please feel free to give an example of use case.

**Comment by becjit** [comment]
Hi,
    Let me try to break it down to you. What I followed from your example
that onLoadMore both triggers and fetches the data. In your example when it
is scrolled onLoadMore will get called with the offset and it will return
the data.

Now imagine there is a state management solution like bloc where triggering
the event and emitting the state is decoupled.
In short using onLoadMore you can trigger an event to get more data which
in turn trigger a state change.

Now my question is where do I listen for the state change?

Assuming you are familiar with flutter bloc that state change post
onloadmore trigger could be listened through blocbuilder.

So in short the event trigger and widget renderings as a reaction to that
event is decoupled

So in typical reactive programming where event trigger is decoupled from
event listener I am not too sure we can make it work. It seemed to me
onloadmore handles both event trigger and event listener part i. E. Trigger
the fetch as well as return the result



Hope this clarifies. If needed I can quickly give you a pseudo code.
Thanks.
Abhijit

**Comment by esentis** [comment]
@becjit Thank you for elaborating on this ! 
I think I know what you need, I will work on it when I find time and will get back at you.

**Comment by esentis** [comment]
@becjit I know it took some time, but it is here with `1.3.0` ! 😄 
I have added detailed example with `BloC`.
